src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -0,0 +1,170 @@
+        """Settings class for the Sagemaker orchestrator.
+
+        Returns:
+            The settings class.
+        """
+        return SagemakerOrchestratorSettings
+
+    def prepare_or_run_pipeline(
+        self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> Any:
+        """Prepares or runs a pipeline on Sagemaker.
+
+        Args:
+            deployment: The deployment to prepare or run.
+            stack: The stack to run on.
+
+        Returns:
+            The result of the pipeline run.
+        """
+        if deployment.schedule:
+            logger.warning(
+                "The Sagemaker Orchestrator currently does not support the "
+                "use of schedules. The `schedule` will be ignored "
+                "and the pipeline will be run immediately."
+            )
+
+        orchestrator_run_name = get_orchestrator_run_name(
+            pipeline_name=deployment.pipeline.name
+        ).replace("_", "-")
+
+        session = sagemaker.Session(default_bucket=self.config.bucket)
+        image_name = deployment.pipeline.extra[ORCHESTRATOR_DOCKER_IMAGE_KEY]
+        execution_role = (
+            self.config.processor_role or sagemaker.get_execution_role()
+        )
+
+        sagemaker_steps = []
+        for step_name, step in deployment.steps.items():
+            command = StepEntrypointConfiguration.get_entrypoint_command()
+            arguments = StepEntrypointConfiguration.get_entrypoint_arguments(
+                step_name=step_name
+            )
+            entrypoint = command + arguments
+
+            step_settings = cast(
+                SagemakerOrchestratorSettings, self.get_settings(step)
+            )
+            execution_role = step_settings.execution_role or execution_role
+
+            processor = sagemaker.processing.Processor(
+                role=execution_role,
+                image_uri=image_name,
+                instance_count=1,
+                sagemaker_session=session,
+                instance_type=step_settings.instance_type,
+                entrypoint=entrypoint,
+                base_job_name=orchestrator_run_name,
+                env={
+                    ENV_ZENML_SAGEMAKER_RUN_ID: ExecutionVariables.PIPELINE_EXECUTION_ARN,
+                },
+                volume_size_in_gb=step_settings.volume_size_in_gb,
+                max_runtime_in_seconds=step_settings.max_runtime_in_seconds,
+            )
+
+            sagemaker_step = ProcessingStep(
+                name=step.config.name,
+                processor=processor,
+                depends_on=step.spec.upstream_steps,
+            )
+            sagemaker_steps.append(sagemaker_step)
+
+        # construct the pipeline from the sagemaker_steps
+        pipeline = Pipeline(
+            name=orchestrator_run_name,
+            steps=sagemaker_steps,
+            sagemaker_session=session,
+        )
+
+        pipeline.create(role_arn=execution_role)
+        pipeline.start()
```

Comment:
Given that @stefannica suggested we need a "synchronous" flag here to also do a `pipeline.wait()` 

Source: https://sagemaker.readthedocs.io/en/stable/workflows/pipelines/sagemaker.workflow.pipelines.html#sagemaker.workflow.pipeline._PipelineExecution.wait

Another cool add-on would be to stream the logs but thats maybe too far for now (unless you find it in the sagemaker sdk)

Reply:
A super simple pipeline of 2 steps adding two integers together takes about 17 or 18 minutes to run. Not sure it's a good experience for a ZenML user to add this. The maximum for that `.wait()` method would be 60*30 seconds (so 30 mins) which would not be enough for many real workflows, I think.

Reply:
Btw you could also not wait by setting the `synchronous` flag to be false when registering the stack component

Reply:
I agree, the synchronous flag should definitely be `False` by default and it should mostly be enabled by tests, not too many other reasonable use-cases for that

Reply:
It seems the slowness is just a thing with these Processing jobs. [This](https://stackoverflow.com/questions/71256518/delay-in-executing-sagemaker-sklearn-processing-sklearnprocessor-run-job-in-sag) suggests that the two things to reduce instance load time is to reduce the image (not really possible for us) and reduce data import. I'll see if adding a dockerignore file for my test pipeline improves things significantly, but I think the 5 minute coldstart time to provision the processor instance may be something we're stuck with for this orchestrator. I'll keep looking, though.

Reply:
Hello,
the processor class of Sagemaker is not for me the reason, it maybe related to your AWS region, instance type, etc.
but not 15 minutes to start the instance for any reason.